onvm/onvm_mgr/onvm_nf.c

```diff
@@ -122,19 +122,25 @@ onvm_nf_check_status(void) {
                         case MSG_NF_STARTING:
                                 nf = (struct onvm_nf_info *)msg->msg_data;
                                 if (onvm_nf_start(nf) == 0) {
-                                        onvm_stats_add_event("NF Starting", nf);
+                                        gen_event_nf_info("NF Starting", nf);
                                 }
                                 break;
                         case MSG_NF_READY:
                                 nf = (struct onvm_nf_info *)msg->msg_data;
                                 if (onvm_nf_ready(nf) == 0) {
-                                        onvm_stats_add_event("NF Ready", nf);
+                                        gen_event_nf_info("NF Ready", nf);
                                 }
                                 break;
                         case MSG_NF_STOPPING:
                                 nf = (struct onvm_nf_info *)msg->msg_data;
                                 if (onvm_nf_stop(nf) == 0) {
-                                        onvm_stats_add_event("NF Stopping", nf);
+                                        gen_event_nf_info("NF Stopping", nf);
+
+                                        /* Cleanup the allocated tag */
+                                        if (nf->tag) {
```

Comment:
tag cleanup is already in onvm_nf_stop, remove this

Reply:
well at least in the nf tag pr, not sure why its removed here

Reply:
I removed it from onvm_nf_stop. The reason is because when calling the json for the react app, on cleanup, when you call `onvm_nf_stop`, nf->tag is NULL. Thus, there is no way of knowing which nf (on the web side) needs to be removed from the display. So I just called it after the event info, as the only time `onvm_nf_stop` is called, is here.

Reply:
Not sure if I understand, why would the tag be required? you have the data from the onvm_nf_info right? How did we stop this before because tags were broken

Reply:
Also is actually dangerous anyway as we're freeing the memory and then trying to use it again.

I propose we rewrite this a little, add a new event macro ONVM_EVENT_NF_STOPPING, and do the message will just have the instance_id value of the NF that was stopped(save instance id as a temp value before calling stop). Does that make sense?

Reply:
It's because Aaron's code assumes that there is no tag at all. So in the code, `NF <instance_id>` is just passed around in the js. But now, I fixed the javascript so that we pass around the tag, so we can see it on every page (like a new id). But when we're cleaning up, in [Stats NF conditonal](https://github.com/sdnfv/openNetVM/pull/101/files?file-filters%5B%5D=.c#diff-048e828d7a71a5d733c5804ba7a851e2R296), we pass the javascript `NF` as the name, instead of `nf->tag` if tag doesn't exist. So on start, an NF with tag will get a custom type, but on termination, it will be falsely called with `NF <instance_id>`, so we can't find it.

Reply:
Tag for id seems like a bad idea though, its not unique

Reply:
The uniqueness comes with `<tag> <instance_id>`

Reply:
Yeah but tag is just a pretty name, instance id is what we should actually use. Is there a benefit of including the tag there that I'm missing? Should we just pass it as extra data?

And as this usage in unsafe we need to rewrite this a little, do you agree with my earlier comment on that?

Reply:
Ya I agree and I'm working on changing it now. I'll figure out a way to fix the js code to work with it